lib/parsers/package-lock-parser.ts

```diff
@@ -128,63 +141,50 @@ export class PackageLockParser implements LockfileParser {
     Once completed for all nodes in a cycle, original cyclic nodes can
     be removed.
   */
-
   private removeCycle(
-    cycle: string[], depMap: DepMap, depGraph: graphlib.Graph): {[cycleStart: string]: string} {
-
-    function walkCycleRec(node, traversed, currentCycle, nodeCopy) {
-      // 2. For every cyclic dependency of entry node...
-      const edgesToProcess = (depGraph.inEdges(node) as graphlib.Edge[])
-        .filter((e) => _.includes(currentCycle, e.v));
-      for (const edge of edgesToProcess) {
-      // ... create a duplicate of the dependency...
-        const copyName = edge.v + uuid();
-      // ...and connect it with the duplicated entry node
-        depGraph.setEdge(copyName, nodeCopy);
-        // metadata update
-        depMap[copyName] = _.cloneDeep(depMap[edge.v]);
-        if (_.includes(traversed, edge.v)) {
-          // 3.a If edge goes to already-visited dependency, end of cycle is found;
-          // update metadata and do not continue traversing
-          depMap[copyName].cyclic = true;
-        } else {
-          // 3.b Follow the edge and repeat the process, storing visited dependency-paths
-          // metadata update
-          walkCycleRec(edge.v, [...traversed, node], currentCycle, copyName);
-          // All non-cyclic dependencies of duplicated node need to be updated.
-          const edgesToCopy = (depGraph.inEdges(edge.v) as graphlib.Edge[])
-            .filter((e) => !_.includes(currentCycle, e.v));
-          for (const edgeToCopy of edgesToCopy) {
-            depGraph.setEdge(edgeToCopy.v, copyName);
-          }
-        }
-      }
-    }
+    cycle: string[], depMap: DepMap, depGraph: graphlib.Graph): CycleStartMap {
 
-    const cycleStarts = {};
+    const cycleStarts: CycleStartMap = {};
     // For every node in a cycle:
     for (const start of cycle) {
       // 1. Create a uniqe duplicate of entry node (without edges)
-      const newNode = start + uuid();
-      cycleStarts[start] = newNode;
-      depMap[newNode] = depMap[start];
-      // update depMap with new node
-      depGraph.setNode(newNode);
-      walkCycleRec(start, [], cycle, newNode);
+      const clonedNode = this.cloneNodeWithoutEdges(start, depMap, depGraph);
+      cycleStarts[start] = clonedNode;
+
+      /* FUNCTION DEFINITION
+        To keep an order of algorithm steps readable, function is defined on-the-fly
+        Arrow function is used for calling `this` without .bind(this) in the end
+      */
+      const acyclicDuplicationRec = (node, traversed, currentCycle, nodeCopy) => {
```

Comment:
this is great - and it will be even more readable if this definition moves 2 lines up, outside of the `for` loop. This way the algorithm can be read more outside-in.